src/osd/OpRequest.cc

```diff
@@ -74,14 +74,14 @@ void OpRequest::_dump(Formatter *f) const
       f->dump_string("event", i->str);
       f->dump_stream("time") << i->stamp;
 
-      auto i_next = i + 1;
+      double duration = 0;
 
-      if (i_next < events.end()) {
-	f->dump_float("duration", i_next->stamp - i->stamp);
-      } else {
-	f->dump_float("duration", events.rbegin()->stamp - get_initiated());
+      if (i != events.begin()) {
+        auto i_prev = i - 1;
+        duration = i->stamp - i_prev->stamp;
```

Comment:
is it always positive? or you can use `std::abs(i->stamp - i_prev->stamp)`

Reply:
It should be :) but i think we should see negative values in case they occur, making them absolute doesn't seem right since that means the event order is messed up.

Reply:
Actually, it would underflow and produce something like `"duration": 4294967295.9999957`, and is pretty common for `"event": "header_read"`